CopilotKit/packages/runtime/src/graphql/resolvers/copilot.resolver.ts

```diff
@@ -28,9 +91,33 @@ export class CopilotResolver {
     if (properties) {
       ctx._copilotkit.properties = { ...ctx._copilotkit.properties, ...properties };
     }
+
+    let copilotCloudPublicApiKey: string | null = null;
+
+    if (data.cloud) {
+      const key = ctx.request.headers.get("x-copilotcloud-public-api-key");
+      if (key) {
+        copilotCloudPublicApiKey = key;
+      } else {
+        throw new GraphQLError("X-CopilotCloud-Public-API-Key header is required");
+      }
+    }
+
     const copilotRuntime = ctx._copilotkit.runtime;
     const serviceAdapter = ctx._copilotkit.serviceAdapter;
-    const responseStatus = new Subject<typeof ResponseStatusUnion>();
+    const responseStatus$ = new Subject<typeof ResponseStatusUnion>();
+    const interruptStreaming$ = new Subject<void>();
```

Comment:
Since `interruptStreaming$` is used by multiple subscribers, shouldn't it be a `ReplaySubject`?